tests/operators/test_python.py

```diff
@@ -311,6 +315,350 @@ def func(**context):
+        @task_decorator
+        def __do_run():
+            return 4
+
+        with self.dag:
+            __do_run()
+            do_runs = [__do_run.copy() for _ in range(20)]
+
+        assert do_runs[-1].task_id == '__do_run__20'
+
+    def test_dict_outputs(self):
+        """Tests pushing multiple outputs as a dictionary"""
+
+        @task_decorator(multiple_outputs=True)
+        def return_dict(number: int):
+            return {
+                'number': number + 1,
+                43: 43
+            }
+
+        test_number = 10
+        with self.dag:
+            return_dict(test_number)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        return_dict.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+
+        ti = dr.get_task_instances()[0]
+        assert ti.xcom_pull(key='number') == test_number + 1
+        assert ti.xcom_pull(key='43') == 43
+        assert ti.xcom_pull() == {'number': test_number + 1, '43': 43}
+
+    def test_tuple_outputs(self):
+        """Tests pushing multiple outputs as tuple"""
+
+        @task_decorator(multiple_outputs=True)
+        def return_tuple(number: int):
+            return number + 1, 43
+
+        test_number = 10
+        with self.dag:
+            return_tuple(test_number)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        return_tuple.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+
+        ti = dr.get_task_instances()[0]
+        assert ti.xcom_pull(key='0') == test_number + 1
+        assert ti.xcom_pull(key='1') == 43
+        assert ti.xcom_pull() == [test_number + 1, 43]
+
+    def test_list_outputs(self):
+        """Tests pushing multiple outputs as list"""
+
+        @task_decorator(multiple_outputs=True)
+        def return_tuple(number: int):
+            return [number + 1, 43]
+
+        test_number = 10
+        with self.dag:
+            return_tuple(test_number)
+
+        dr = self.dag.create_dagrun(
+            run_id=DagRunType.MANUAL.value,
+            start_date=timezone.utcnow(),
+            execution_date=DEFAULT_DATE,
+            state=State.RUNNING
+        )
+
+        return_tuple.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+
+        ti = dr.get_task_instances()[0]
+        assert ti.xcom_pull(key='0') == test_number + 1
+        assert ti.xcom_pull(key='1') == 43
+        assert ti.xcom_pull() == [test_number + 1, 43]
+
+    def test_xcom_arg(self):
+        """Tests that returned key in XComArg is returned correctly"""
+
+        @task_decorator
+        def add_2(number: int):
+            return number + 2
+
+        @task_decorator
+        def add_num(number: int, num2: int = 2):
```

Comment:
Do defaults work? We should add a test for that I think.

Reply:
They do. I can add a test.